faucet/valve.py

```diff
@@ -1361,6 +1366,23 @@ def get_config_dict(self):
             'acls': acls_dict,
             }
 
+    def flow_timeout(self, table_id, match):
+        eth_src = None
+        match_oxm_fields = match.to_jsondict()['OFPMatch']['oxm_fields']
+        if table_id == self.dp.eth_src_table:
```

Comment:
The dst rules cannot expire before the src rules. If a host keeps sending without receiving, the src rule stays. The controller then refresh the dst rule for every "expire" period.
If the host may receive without sending, then the controller expires it as normal.